src/components/Datatable/Datatable.tsx

```diff
@@ -134,17 +140,28 @@ const Datatable = <D extends Record<string, unknown>>({
     [onDataFetch, appliedFilters, searchQuery],
   );
 
+  const handleOnSuggestionsFetch = useCallback(
+    async (query: string) => {
+      setFilterSuggestions(await onSuggestionsFetch(query));
+    },
+    [onSuggestionsFetch],
+  );
+
   const handleOnSearch = useCallback(
-    (query: string) => {
-      setSearchQuery(query);
-      onDataFetch({
-        pageSize: defaultPageSize,
-        pageIndex: 0,
-        query,
-        filters: appliedFilters,
-      });
+    async (queryValue: string) => {
+      setSearchQuery(queryValue);
+      if (controlsConfig.searchConfig?.hasSuggestions === true) {
```

Comment:
Do not use `controlsConfig` (passed as prop into Datatable component), use only the merged configs from L88 and L96, this would lead to unnecessary bugs like before

Reply:
fixed